### PR TITLE
#115 setup react query

### PR DESCRIPTION
**Pull request by ninman27**
- Set up React Query's QueryClient and provided it at the top level of the app in the Layout
- Created a custom Hook for fetching data
- Implemented error handling and loading states
- Created a simple component Users folder (Users.tsx)

### CODE REVIEW

**Review by edichoska — DISMISSED** [review]
@ninman27 can you also include useMutaion?

File: components/module-components/users/Users.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import getUsers from '../../../app/getUsers';
+
+const Users = () => {
+  const { data, isLoading, isError } = useQuery({
+    // eslint-disable-next-line no-return-await
+    queryFn: async () => await getUsers(),
+    queryKey: ['users'],
+  });
+
+  if (isLoading) return <div>Loading</div>;
```
Comment (edichoska) [inline components/module-components/users/Users.tsx:13]: can we display here the global loading component we have?

**Comment by edichoska** [comment]
@ninman27 I keep running into this error:

**Comment by ninman27** [comment]
@edichoska I can't open the image...

**Comment by edichoska** [comment]
Screenshot 2024-04-16 at 00 09 56

File: app/getUsers.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+async function getData() {
+  try {
+    const response = await fetch('https://jsonplaceholder.typicode.com/users');
+    const data = response.json();
```
Comment (stoimenov95) [inline app/getUsers.tsx:4]: @ninman27 We can return the response.json directly instead of creating the data variable that is only used once.

**Review by stoimenov95 — CHANGES_REQUESTED** [review]

**Comment by stoimenov95** [comment]
> Screenshot 2024-04-16 at 00 09 56

I don't experience this problem...

**Review by stoimenov95 — CHANGES_REQUESTED** [review]
Except for some unnecessary variables, everything else looks awesome @ninman27, great job!

**Review by stoimenov95 — APPROVED** [review]
🥳

**Pull request merged**